samples/DotNet/MSI/EventHubsMSIDemoWebApp/EventHubsMSIDemoWebApp/EventHubsMSIDemo.aspx.cs

```diff
@@ -0,0 +1,81 @@
+﻿using Microsoft.ServiceBus;
+using Microsoft.ServiceBus.Messaging;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Web;
+using System.Web.UI;
+using System.Web.UI.WebControls;
+
+// Always add app to IAM roles
+// Don't use on deployment slots but only on root
+namespace EventHubsMSIDemoWebApp
+{
+    public partial class EventHubsMSIDemo : System.Web.UI.Page
+    {
+        protected void Page_Load(object sender, EventArgs e)
+        {
+
+        }
+
+        protected void btnSend_Click(object sender, EventArgs e)
+        {
+            MessagingFactorySettings messagingFactorySettings = new MessagingFactorySettings
+            {
+                TokenProvider = TokenProvider.CreateManagedServiceIdentityTokenProvider(ServiceAudience.EventHubsAudience),
+                TransportType = TransportType.Amqp
+            };
+
+            // TODO - Remove after backend is patched with the AuthComponent open fix
+            messagingFactorySettings.AmqpTransportSettings.EnableLinkRedirect = false;
```

Comment:
Redirect is on the receiver only, doesn't apply to send. You can remove this.

Reply:
Done